src/models/contracts/objectExplorer/nodeInfo.ts

```diff
@@ -0,0 +1,51 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+/**
+ * Information describing a Node in the Object Explorer tree.
+ * Contains information required to display the Node to the user and
+ * to know whether actions such as expanding children is possible
+ * the node
+ */
+export class NodeInfo {
+
+    /**
+     * Path identifying this node: for example a table will be at ["server", "database", "tables", "tableName"].
+     * This enables rapid navigation of the tree without the need for a global registry of elements.
+     * The path functions as a unique ID and is used to disambiguate the node when sending requests for expansion.
+     * A common ID is needed since processes do not share address space and need a unique identifier
+     */
+    public nodePath: string;
+
+    /**
+     * The type of node - for example Server, Database, Folder, Table
+     */
+    public nodeType: string;
+
```

Comment:
How is this set? It would be greatly preferential to have this be a string enum since these should all be known types

Reply:
This comes from the tools service